### PR TITLE
Merge branch 'v3.6.1-taobao' into v3.6.3

### PR DESCRIPTION
**Pull request by yyycggg**
Re: #

### Changelog

* Support Taobao

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by PPpro — APPROVED** [review]

File: platforms/minigame/common/engine/AssetManager.js [hunk 1]
```diff
@@ -23,7 +23,9 @@ function downloadScript (url, options, onComplete) {
         onComplete && onComplete(new Error('Can not load remote scripts'));
     }
     else {
-        require('../../../' + url);
+        if (sys.platform !== sys.Platform.TAOBAO_CREATIVE_APP) {
```
Comment (minggo) [inline platforms/minigame/common/engine/AssetManager.js:27]: Please add comment.

File: platforms/minigame/common/engine/AssetManager.js [hunk 2]
```diff
@@ -240,7 +242,9 @@ function downloadBundle (nameOrUrl, options, onComplete) {
                 js = `assets/${bundleName}/index.${suffix}js`;
             }
         }
-        require('./' + js);
+        if (sys.platform !== sys.Platform.TAOBAO_CREATIVE_APP) {
```
Comment (minggo) [inline platforms/minigame/common/engine/AssetManager.js:247]: Please add comment

File: platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+let CURR_AMOUNT_WEBSOCKET = 0    // The current number of WEBSOCKET
+
+export default class WebSocket {
+  static CONNECTING = 0 // The connection is not yet open.
+  static OPEN = 1 // The connection is open and ready to communicate.
+  static CLOSING = 2 // The connection is in the process of closing.
+  static CLOSED = 3 // The connection is closed or couldn't be opened.
+
+  binaryType = '' // TODO 更新 binaryType
+  bufferedAmount = 0 // TODO 更新 bufferedAmount
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js]: Don't use chinese.

File: platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  onerror = null
+  onclose = null
+
+  _onMessage = null
+  _onOpen = null
+  _onError = null
+  _onClose = null
+  _isReduced = false
+
+  protocol = '' // TODO 小程序内目前获取不到，实际上需要根据服务器选择的 sub-protocol 返回
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js]: The same as above.

File: platforms/minigame/platforms/taobao/wrapper/builtin/XMLHttpRequest.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+  }
+}
+
+function _changeReadyState(readyState) {
+  this.readyState = readyState
+  _triggerEvent.call(this, 'readystatechange')
+}
+
+export default class XMLHttpRequest extends EventTarget {
+  // TODO 没法模拟 HEADERS_RECEIVED 和 LOADING 两个状态
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/XMLHttpRequest.js]: The same as above.

File: platforms/minigame/platforms/taobao/wrapper/builtin/XMLHttpRequest.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  static OPENED = 1
+  static HEADERS_RECEIVED = 2
+  static LOADING = 3
+  static DONE = 4
+
+  timeout = 0;
+  /*
+   * TODO 这一批事件应该是在 XMLHttpRequestEventTarget.prototype 上面的
+   */
+  onabort = null
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/XMLHttpRequest.js:32]: The same as above.

File: platforms/minigame/platforms/taobao/wrapper/builtin/XMLHttpRequest.js [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    return Object.keys(responseHeader).map((header) => {
+      return `${header}: ${responseHeader[header]}`
+    }).join('\n')
+  }
+
+  getResponseHeader(header) {
+    return _responseHeader.get(this)[header]
+  }
+
+  open(method, url/* async, user, password 这几个参数在小程序内不支持*/) {
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/XMLHttpRequest.js]: the same as above

File: platforms/minigame/platforms/taobao/wrapper/builtin/index.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
+import * as _window from './window'
+import { document } from './document'
+
+var global = $global;
+
+function inject () {
+  // 暴露全局的 canvas
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/index.js]: The same as above

File: platforms/minigame/platforms/taobao/wrapper/builtin/index.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  }
+  _window.removeEventListener = (type, listener) => {
+    _window.document.removeEventListener(type, listener)
+  }
+  _window.dispatchEvent = _window.document.dispatchEvent;
+
+  const { platform } = my.getSystemInfoSync()
+
+  // 开发者工具无法重定义 window
+  if (typeof __devtoolssubcontext === 'undefined' && platform === 'devtools') {
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/index.js:22]: The same as above

File: platforms/minigame/platforms/taobao/wrapper/builtin/navigator.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import { noop } from './util/index.js'
+
+// TODO 需要 my.getSystemInfo 获取更详细信息
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/navigator.js]: The same as above

File: platforms/minigame/platforms/taobao/wrapper/builtin/navigator.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+const ua = `Mozilla/5.0 (${uaDesc}) AppleWebKit/603.1.30 (KHTML, like Gecko) Mobile/14E8301 MicroMessenger/${version} MiniGame NetType/WIFI Language/${language}`;
+
+const navigator = {
+  platform,
+  language: language,
+  appVersion: `5.0 (${uaDesc}) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13B143 Safari/601.1`,
+  userAgent: ua,
+  onLine: true, // TODO 用 my.getNetworkStateChange 和 my.onNetworkStateChange 来返回真实的状态
+
+  // TODO 用 my.getLocation 来封装 geolocation
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/navigator.js]: The same as above

File: platforms/minigame/platforms/taobao/wrapper/fs-utils.js [hunk 12]
```diff
@@ -0,0 +1,262 @@
+/****************************************************************************
+ Copyright (c) 2017-2019 Xiamen Yaji Software Co., Ltd.
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/fs-utils.js]: error date

File: platforms/minigame/platforms/taobao/wrapper/fs-utils.js [hunk 13]
```diff
@@ -0,0 +1,262 @@
...
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ****************************************************************************/
+// TODO: verify the my API
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/fs-utils.js]: Is `TODO` needed?

File: platforms/minigame/platforms/taobao/wrapper/builtin/Image.js [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+function Image () {
+    // empty constructor
+}
+
+// NOTE: Proxy not supported on iOS 8 and 9.
+// let ImageProxy = new Proxy(Image, {
+//     construct (target, args) {
+//         let img =  screencanvas.createImage();
+//         if (!img.addEventListener) {
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/Image.js]: Image can not work now?

Comment (yyycggg) [reply platforms/minigame/platforms/taobao/wrapper/builtin/Image.js]: The image can work now . The previous code has not been removed, it has just been commented out.

Comment (minggo) [reply platforms/minigame/platforms/taobao/wrapper/builtin/Image.js]: If so, then please remove it and add comment.

Comment (yyycggg) [reply platforms/minigame/platforms/taobao/wrapper/builtin/Image.js]: OK.

**Review by minggo — APPROVED** [review]

**Pull request merged**